recipes/pcre2/all/conanfile.py

```diff
@@ -98,6 +98,8 @@ def generate(self):
         tc.variables["PCRE2_BUILD_PCRE2_32"] = self.options.build_pcre2_32
         tc.variables["PCRE2_SUPPORT_JIT"] = self.options.support_jit
         tc.variables["PCRE2GREP_SUPPORT_CALLOUT_FORK"] = self.options.get_safe("grep_support_callout_fork", False)
+        # Fix for error: 'for' loop initial declarations are only allowed in C99 or C11 mode
+        tc.variables["CMAKE_C_STANDARD"] = "99"
```

Comment:
According to https://github.com/PCRE2Project/pcre2/pull/193, it's a recent change, so it shouldn't be forced globally like this regardless of pcre2 version. I would recommend to patch impacted versions instead.

Reply:
Great point, thanks for the feedback :+1: 

I've replaced this with a patch for the 10.42 version instead.

Reply:
Modified the patch so that it now cleanly applies to both 10.40 and 10.42